gbpservice/nfp/configurator/agents/generic_config.py

```diff
@@ -0,0 +1,468 @@
+            """For other events, irrespective of result send notification"""
+            notification_data = self._prepare_notification_data(ev, result)
+            self._notification(notification_data)
+
+    def _prepare_notification_data(self, ev, result):
+        """Prepare notification data as expected by config agent
+
+        :param ev: event object
+        :param result: result of the handled event
+
+        Returns: notification_data
+
+        """
+        kwargs = ev.data.get('kwargs')
+        request_info = kwargs['request_info']
+        del kwargs['request_info']
+        context = ev.data.get('context')
+
+        # Retrieve notification and remove it from context. Context is used
+        # as transport from batch processing function to this last event
+        # processing function. To keep the context unchanged, delete the
+        # notification_data before invoking driver API.
+        notification_data = context.get('notification_data')
+        del context['notification_data']
+        resource = context.get('resource')
+        del context['resource']
+
+        msg = {'receiver': gen_cfg_const.ORCHESTRATOR,
+               'resource': resource,
+               'method': ev.id.lower(),
+               'kwargs': [
+                          {
+                           'context': context,
+                           'resource': resource,
+                           'request_info': request_info,
+                           'result': result
+                          }
+                        ]
+               }
+        if not notification_data:
+            notification_data.update(msg)
+        else:
+            data = {
+                    'context': context,
+                    'resource': resource,
+                    'request_info': request_info,
+                    'result': result
+                   }
+            notification_data['kwargs'].append(data)
+        return notification_data
+
+    def poll_event_cancel(self, ev):
+        """Invoked by process framework when poll ev object reaches
+           polling threshold ev.max_times.
+           Finally it Enqueues response into notification queue.
+
+        :praram ev: event object
+
+        Returns: None
+
+        """
+        LOG.error('Poll event cancelled for event %s ' % (ev.data))
+        result = common_const.FAILED
+        notification_data = self._prepare_notification_data(ev, result)
+        self._notification(notification_data)
+
+    @nfp_poll.poll_event_desc(
+                            event=gen_cfg_const.EVENT_CONFIGURE_HEALTHMONITOR,
+                            spacing=5)
+    def handle_configure_healthmonitor(self, ev):
+        """Decorator method called for poll event CONFIGURE_HEALTHMONITOR
+           Finally it Enqueues response into notification queue.
+
+        :param ev: event object
+
+        Returns: None
+
+        """
+        self._process_event(ev)
+
+
+def events_init(sc, drivers, rpcmgr, nqueue):
```

Comment:
Events are specific to a service type. So, the event_init() function should be in agent specific file rather than agent base.
There is possibility of moving load_drivers() and register_service_agent() to agent base and all service specific agents can use them.
Is this what you are suggesting?